### PR TITLE
Add npm prompts package for command line prompts support with other features

### PR DESCRIPTION
**Pull request by OptimBro**
Fixes #<issue_number_goes_here>

> It's a good idea to open an issue first for discussion.

- [ ] Tests pass
- [ ] Appropriate changes to README are included in PR
- [x] Types updated

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
     "globby": "^13.1.1",
     "minimist": "^1.2.5",
     "node-fetch": "^3.2.3",
+    "prompts": "^2.4.2",
```
Comment (antonmedv) [inline package.json:44]: Prompts comes with types?

File: src/globals.d.ts [hunk 2]
```diff
@@ -8,26 +8,28 @@ import {
   globby as _globby,
   nothrow,
   os as _os,
+  prompts as _prompts,
   path as _path,
   question,
   sleep,
   which as _which,
-} from './index'
+} from './index';
```
Comment (antonmedv) [inline src/globals.d.ts:16]: No need to add it here.

File: src/index.mjs [hunk 3]
```diff
@@ -25,15 +25,16 @@ import chalk from 'chalk'
 import YAML from 'yaml'
 import minimist from 'minimist'
 import psTreeModule from 'ps-tree'
+import prompts from 'prompts';
 
-export {chalk, fs, os, path, YAML, which}
-export const sleep = promisify(setTimeout)
-export const argv = minimist(process.argv.slice(2))
+export { chalk, fs, os, path, YAML, which, prompts };
```
Comment (antonmedv) [inline src/index.mjs:30]: Extra space not needed.

**Review by antonmedv — CHANGES_REQUESTED** [review]

File: src/index.mjs (hunk 3 above)
Comment (OptimBro) [reply src/index.mjs:30]: Fixed

File: src/globals.d.ts (hunk 2 above)
Comment (OptimBro) [reply src/globals.d.ts:16]: got it

File: package.json (hunk 1 above)
Comment (OptimBro) [reply package.json:44]: Yes, it comes with types. I am not sure about typescript but it also has this: https://www.npmjs.com/package/@types/prompts

Comment (OptimBro) [reply package.json:44]: image

yes, it comes with types. Auto complete working fine on vscode as well.

Comment (antonmedv) [reply package.json:44]: So no?

**Review by OptimBro — COMMENTED** [review]
Changes uploaded

File: package.json (hunk 1 above)
Comment (OptimBro) [reply package.json:44]: > So no?

It comes with types.

**Comment by antongolub** [comment]
Let's apply lingting as a separate commit / PR.

**Comment by OptimBro** [comment]
> Let's apply linter in a separate commit / PR.

You mean to say reupload changes in new PR? Sorry, I am new to this.

**Comment by antongolub** [comment]
I suggest to revert formatting, commas, etc to make your change atomic — to bring just the new `promts` API with tests and docs.

**Comment by antongolub** [comment]
@antonmedv,

`prompts` → `kleur`. `chalk` is +/- the same. 2 impls of console colorer is too much imo.

**Comment by OptimBro** [comment]
> I suggest to revert formatting, commas, etc to make your change atomic — to bring just the new `promts` API with tests and docs.

Okay, I got it. It was done by my editor plugins. I will reupload changes.

**Comment by OptimBro** [comment]
> @antonmedv,
> 
> `prompts` → `kleur`. `chalk` is +/- the same. 2 impls of console colorer is too much imo.

No, its not for console colorer. It is meant for command line interactivity with various use cases. I already use chalk when it comes for coloring. `prompts` is useful when I want use to choose one option from multiple option, or choose multiple option etc etc.

**Comment by antongolub** [comment]
image
Just thought if we land this change we could also wrap kluer with chalk-compatible facade, and throw the last from deps.

**Comment by OptimBro** [comment]
> image
> 
> Just thought if we land this change we could also wrap kluer with chalk-compatible facade, and throw the last from deps.

Ahh, got it. I will try it myself here and will do a pull request :)

**Comment by antonmedv** [comment]
We can just keep kleur if we decide on including prompts. No harm in that.

**Comment by OptimBro** [comment]
> We can just keep kleur if we decide on including prompts. No harm in that.

Sure, I'll test and do the needful. For time being, I have created a new [pull request](https://github.com/google/zx/pull/381) which just adds prompts package.

I'll test and replace chalk with kleur in other pull request ^^

Thank you

**Comment by antonmedv** [comment]
> I'll test and replace chalk with kleur in other pull request ^^

nope 👎

**Comment by antongolub** [comment]
What we gonna with zx deps size in future? `prompts` requires +187 kB. What's the limit?

**Comment by antonmedv** [comment]
Wow, probably we don;t want to add prompts, if it's size is 187 kB.

**Comment by OptimBro** [comment]
> Wow, probably we don;t want to add prompts, if it's size is 187 kB.

Okay, so what are your thoughts about [inquirer](https://www.npmjs.com/package/inquirer) ?

It is  just 86 KB in size, and also uses chalk behind the scenes. This can be added I think :)

image

**Comment by antonmedv** [comment]
I’m thinking of not adding such library for now.

**Comment by OptimBro** [comment]
Sure, thanks :)

I'll use my fork instead then.

**Pull request closed without merging**